pkg/utils/kubelessutil.go

```diff
@@ -430,13 +431,51 @@ func populatePodSpec(funcObj *kubelessApi.Function, lr *langruntime.Langruntimes
 			*compContainer,
 		)
 	}
+
+	// mount volumes with init container secrets specified in runtime configuration
+	lr.ReadConfigMap()
+	for i := 0; i < len(result.InitContainers); i++ {
+		secrets, err := lr.GetInitContainerSecrets(funcObj.Spec.Runtime, result.InitContainers[i].Name)
+		if err != nil {
+			return fmt.Errorf("Unable to fetch init container secrets for runtime %s at phase %s: %v", funcObj.Spec.Runtime, result.InitContainers[i].Name, err)
+		}
+		for _, secret := range secrets {
+			// add volume if not available in the pod spec already
+			var found bool
+			for _, vol := range result.Volumes {
+				if vol.Name == secret.Name && (vol.Secret == nil || vol.Secret.SecretName != secret.Name) {
+					return fmt.Errorf("Unable to add volume for secret %s, volume already defined %#v", secret.Name, vol)
+				}
+				if vol.Name == secret.Name && vol.Secret != nil && vol.Secret.SecretName == secret.Name {
+					found = true
+					break
+				}
+			}
+			if !found {
+				result.Volumes = append(result.Volumes, v1.Volume{
+					Name: secret.Name,
+					VolumeSource: v1.VolumeSource{
+						Secret: &v1.SecretVolumeSource{SecretName: secret.Name},
+					},
+				})
+			}
+
+			// add volume mount to the init container
+			result.InitContainers[i].VolumeMounts = append(result.InitContainers[i].VolumeMounts, v1.VolumeMount{
+				Name:      secret.Name,
+				ReadOnly:  true,
+				MountPath: "/" + secret.Name,
```

Comment:
you need to use an unused subpath like `"/etc/secrets/" + secret.Name` so you avoid issues with people using as secret name `var` or `run`

Reply:
Hi @andresmgot , I agree completely, mounting in the root is not ideal - I went for it as a first iteration + AFAIK `serverless-kubeless` plugin  is using this approach.

I am thinking of the following options:
1. `/var/run/secrets/{{secret.name}}`
1. `/var/run/secrets/kubeless.io/{{secret.name}}`
1. `/opt/secrets/{{secret.name}}`
1. `/opt/secrets/kubeless.io/{{secret.name}}`

My choice would be 2. Seems consistent with what I see as mounted volumes with secrets in kubernetes.

What do you think?

Reply:
Changed mount path to `/var/run/secrets/kubeless.io/{{secret.name}}`.